src/tests/unit/tests/popup/components/launch-pad.test.tsx

```diff
@@ -108,6 +108,6 @@ describe('LaunchPad', () => {
             </div>
         );
 
-        expect(shallow(testObject.render()).debug()).toEqual(shallow(expected).debug());
+        expect(render(testObject.render()).debug()).toEqual(render(expected).debug());
```

Comment:
`.debug()` just returns `undefined` so this will always be true. Using `asFragment()` instead will get the actual markup. You'll need to mock `ExternalLink` and `LaunchPadItemRow` so that the elements actually match because of generated ids.

```suggestion
        expect(render(testObject.render()).asFragment()).toEqual(render(expected).asFragment());
```

Reply:
Hi i have made changes, please re-review